### ISSUE TITLE
`memcpy` and `malloc` instead of `new` and loop assignments

### ISSUE THREAD

**Issue by turboelk** [Defavlt/libslw#2]
As used in [SLW/cstring.h](https://github.com/Defavlt/SLW/blob/98575c5e3972534363790fd6eef3f947a8cbf577/include/slw/cstrings.h#L21), line 21-22:

``` C++
char* substr = ( char* ) malloc( 1 + length * sizeof( char ) );
memcpy( substr, src + start, length + 1 );
```

could possibly be replaced with something like

``` C++
char *substr = (char *)new(nothrow) char[ 1 + length ];

for(size_t i = start; i < length; ++i)
    substr[i] = src[i];
```

or even outright `std::string`'s.

Considerations:
- How much of the api needs to change?
  - for the better, or
  - for worse?
- Is it worth the effort?

**Comment by Defavlt** [comment]
Fixed by cecf8fd5545e4a7723c8101fb0898bc0d600d6cf

**Comment by turboelk** [comment]
Fixed by cecf8fd5545e4a7723c8101fb0898bc0d600d6cf
